scylla-cdc/src/log_reader_builder.rs

```diff
@@ -0,0 +1,310 @@
+    }
+
+    pub fn keyspace(mut self, keyspace: &str) -> Self {
+        self.keyspace = Some(keyspace.to_string());
+        self
+    }
+
+    pub fn table_name(mut self, table_name: &str) -> Self {
+        self.table_name = Some(table_name.to_string());
+        self
+    }
+
+    pub fn start_timestamp(mut self, start_timestamp: chrono::Duration) -> Self {
+        self.start_timestamp = start_timestamp;
+        self
+    }
+
+    pub fn end_timestamp(mut self, end_timestamp: chrono::Duration) -> Self {
+        self.end_timestamp = Some(end_timestamp);
+        self
+    }
+
+    pub fn window_size(mut self, window_size: Duration) -> Self {
+        self.window_size = window_size;
+        self
+    }
+
+    pub fn safety_interval(mut self, safety_interval: Duration) -> Self {
+        self.safety_interval = safety_interval;
+        self
+    }
+
+    pub fn sleep_interval(mut self, sleep_interval: Duration) -> Self {
+        self.sleep_interval = sleep_interval;
+        self
+    }
+
+    pub fn readers(mut self, readers: Vec<Arc<StreamReader>>) -> Self {
+        self.readers = readers;
+        self
+    }
+
+    pub fn consumer_factory(mut self, consumer_factory: Arc<dyn ConsumerFactory>) -> Self {
+        self.consumer_factory = Some(consumer_factory);
+        self
+    }
+
+    pub async fn build(self) -> anyhow::Result<(CDCLogReader, RemoteHandle<anyhow::Result<()>>)> {
+        if self.table_name.is_none() {
+            return Err(anyhow!(NewCDCLogReaderError::TableName.as_str()));
+        }
+
+        if self.keyspace.is_none() {
+            return Err(anyhow!(NewCDCLogReaderError::KeyspaceName.as_str()));
+        }
+
+        if self.session.is_none() {
+            return Err(anyhow!(NewCDCLogReaderError::Session.as_str()));
+        }
+
+        if self.consumer_factory.is_none() {
+            return Err(anyhow!(NewCDCLogReaderError::ConsumerFactory.as_str()));
+        }
+
+        let mut cdc_reader_worker = CDCReaderWorker {
+            session: self.session,
+            keyspace: self.keyspace,
+            table_name: self.table_name,
+            start_timestamp: self.start_timestamp,
+            end_timestamp: self.end_timestamp,
+            window_size: self.window_size,
+            safety_interval: self.safety_interval,
+            sleep_interval: self.sleep_interval,
+            readers: self.readers,
+            end_timestamp_receiver: self.end_timestamp_receiver,
+            consumer_factory: self.consumer_factory,
+        };
+
+        let (fut, handle) = async move { cdc_reader_worker.run().await }.remote_handle();
+        tokio::task::spawn(fut);
+        let printer = CDCLogReader::new(self.end_timestamp_sender);
+
+        Ok((printer, handle))
+    }
+}
+
+impl Default for CDCLogReaderBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+struct CDCReaderWorker {
+    session: Option<Arc<Session>>,
+    keyspace: Option<String>,
+    table_name: Option<String>,
```

Comment:
Why did you change the field types to `Option` in the reader worker? The reader builder is supposed to validate that they are not `None`, so no need to keep them as `Option`s in the reader worker. This only introduces completely unnecessary unwrapping - future readers of the code will wonder why it is needed and if it can actually fail or not.